### ISSUE TITLE
[dc.js migration] The wrong tooltip type is being selected for the chart

### ISSUE THREAD

**Issue by EmmadUsmani** [metabase/metabase#39550]
question/12329
https://metaboat.slack.com/archives/C06G94JTWBS/p1706193082481509

| dc.js | echarts |
|--------|--------|
| <img width="491" alt="Screenshot 2024-03-04 at 11 51 09 AM" src="https://github.com/metabase/metabase/assets/37751258/8cd505d6-8fe4-4b1c-94a6-babf2b37d584"> | <img width="835" alt="Screenshot 2024-03-04 at 11 47 37 AM" src="https://github.com/metabase/metabase/assets/37751258/41b9f4f4-e1bd-4b00-9f8a-471da1a39933"> | 

question/14061
https://metaboat.slack.com/archives/C06G94JTWBS/p1706194432975859

question/14706
https://metaboat.slack.com/archives/C06G94JTWBS/p1706197291970589

**Comment by JesseSDevaney** [comment]
### question/12329
- https://metaboat.slack.com/archives/C06G94JTWBS/p1706193082481509

1. This question had a Bar/Area specific setting (["stackable.stack_type" !== null](https://github.com/metabase/metabase/blob/echarts/frontend/src/metabase/visualizations/lib/settings/graph.js#L273)) previous set on it
2. causing the tooltip type  (["graph.tooltip_type" = "series_comparison"](https://github.com/metabase/metabase/blob/echarts/frontend/src/metabase/visualizations/lib/settings/graph.js#L337)) to be different from the intended one for Line Charts (`"default"`)
3. This is causing the [stackedTooltipModel](https://github.com/metabase/metabase/blob/echarts/frontend/src/metabase/visualizations/visualizations/CartesianChart/events.ts#L255-L258) to be non-null
4. Because it is non-null, the `<StackedDataTooltip />` gets used instead of `<KeyValuePairChartTooltip />` ([code](https://github.com/metabase/metabase/blob/echarts/frontend/src/metabase/visualizations/components/ChartTooltip/ChartTooltip.tsx#L28-L30))

In Summary:
- This bug is more coming from a differing charts setting carrying-over into the line charts territory
